src/node_url.cc

```diff
@@ -407,29 +407,62 @@ int64_t ParseIPv4Number(const char* start, const char* end) {
   return strtoll(start, nullptr, R);
 }
 
+// https://url.spec.whatwg.org/#ipv4-number-parser
+bool IsIPv4NumberValid(std::string_view input) {
+  if (input.empty()) {
+    return false;
+  }
+
+  if (input.size() >= 2 && input[0] == '0') {
+    if (input[1] == 'X' || input[1] == 'x') {
+      if (input.size() == 2) {
+        return true;
+      }
+
+      return input.find_first_not_of("0123456789abcdefABCDEF", 2) ==
+             std::string_view::npos;
+    } else {
+      return input.find_first_not_of("01234567", 1) == std::string_view::npos;
+    }
+  }
+
+  return std::all_of(input.begin(), input.end(), ::isdigit);
+}
+
 // https://url.spec.whatwg.org/#ends-in-a-number-checker
-bool EndsInANumber(const std::string& input) {
-  std::vector<std::string> parts = SplitString(input, '.', false);
+bool EndsInANumber(const std::string_view input) {
+  if (input.empty()) {
+    return false;
+  }
+
+  char delimiter = '.';
+  auto pointer_start = input.begin();
```

Comment:
If you use `std::string_view::iterator` it might solve the issues related to std::string_view construct on windows.

Reply:
done!

Reply:
Windows build is still failing : /

```
D:\a\node\node\src\node_url.cc(465,66): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'std::string_view' [D:\a\node\node\libnode.vcxproj]
D:\a\node\node\src\node_url.cc(465,66): message : No constructor could take the source type, or constructor overload resolution was ambiguous [D:\a\node\node\libnode.vcxproj]
  node_util.cc
```

Reply:
I would like to know if I can go with [&(*it);](https://stackoverflow.com/a/19569899) again

Reply:
@addaleax any recommendations?

Reply:
it'd look like this:
`std::string_view(&(*pointer_start), pointer_end - pointer_start));`

Reply:
what i understood is that iterators don't need to be pointers although they could be implemented as one. And some implementations of the STL (MSVC++, Mingw) implement interators with special classes (with no pre-difined casts to pointer), so that we cannot use them directly in the string_view constructor that we are using, which needs a charT*.

then we could use * which is overloaded to do what logically mean: convert the iterator type to its pointed-to object, then pass the address of this &(*it).

https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator
https://stackoverflow.com/questions/32654108/c-stdvectoriterator-is-not-a-pointer-why/64910815#64910815